src/lsm/manifest_log.zig

```diff
@@ -255,14 +309,29 @@ pub fn ManifestLogType(comptime Storage: type) type {
                 const table = &tables_used[entry];
                 assert(table.tree_id >= manifest_log.options.tree_id_min);
                 assert(table.tree_id <= manifest_log.options.tree_id_max);
+                assert(table.address > 0);
 
-                if (manifest.insert_table_extent(table.address, block_reference.address, entry)) {
-                    switch (label.event) {
-                        .insert => manifest_log.open_event(manifest_log, label.level, table),
-                        .remove => manifest.queue_for_compaction(block_reference.address),
-                    }
+                if (label.event == .remove) {
+                    assert(manifest_log.tables_removed.get(table.address) == null);
+
+                    manifest.queue_for_compaction(block_address);
+                    manifest_log.tables_removed.putAssumeCapacity(table.address, {});
```

Comment:
Maybe `fetchPut` / `getOrPut` to avoid the second hashmap lookup for the sake of the assert?